### PR TITLE
[feature branch merge] docs: restructure Getting Started section

### PR DESCRIPTION
**Pull request by qmonnet**
This PR aims at merging the [`feature/gsod-docs-reorg`](https://github.com/cilium/cilium/tree/feature/gsod-docs-reorg) feature branch. The feature branch was created because we did not want to risk leaving Cilium's docs in the middle of a major reorganisation when branching for 1.12. Branching is now finished, the reorganisation is progressing well and will be achieved before 1.13, so the feature branch is no longer necessary.

The feature branch currently contains the commits from these two PRs from @yoyo-go:

- https://github.com/cilium/cilium/pull/20175, rebased without conflict
- https://github.com/cilium/cilium/pull/20244, for which conflicts with the following PRs were addressed:

   - #19379 (File requirements-aks.rst split into two files)
   - #20369 (File host-services.rst removed)
   - #19760 (New files under gettingstarted/servicemesh/)
   - #20228 (New files under gettingstarted/clustermesh/)
   - #20439 (New files under gettingstarted/clustermesh/)
   - #20428 (New refs to update in troubleshooting_servicemesh.rst)
   - #20260 (One ref moved to troubleshooting_clustermesh.rst)

CODEOWNERS was also updated for both commits to reflect the updated paths.

Last commit adds a few fixes regarding to the `gs_guide` and `gs_install` link targets, to address some feedback from Joe.

cc @yoyo-go @xmulligan @lizrice

### CODE REVIEW

**Review by twpayne — APPROVED** [review]
👍 for IPAM (by running `make render-docs` and exploring).

**Comment by qmonnet** [comment]
Netlify preview is here: https://deploy-preview-20681--docs-cilium-io.netlify.app/

(Tam explained that Netlify compares the iterations of a PR; in this case it didn't detect any changes after my latest force-push, which only touched CODEOWNERS, and cancelled the new build.)

File: Documentation/concepts/observability/index.rst
Comment (joestringer) [inline Documentation/concepts/observability/index.rst:18]: For the most part, this PR just moves files around so that seems fine. For some pages like this they appear to be deleted. I can't tell if this is intentional and another corresponding page was added elsewhere, or the reshuffle makes this page redundant, or something else.

File: Documentation/index.rst
Comment (joestringer) [inline Documentation/index.rst:86]: microk8s seems a bit random to just be added under operations here.

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joestringer) [inline Documentation/gettingstarted/k8s-install-default.rst]: Is this link replacement correct?

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joestringer) [inline Documentation/gettingstarted/k8s-install-default.rst]: Does `advancedinstallation` exist? At a glance it doesn't look like this is where the page ended up.

File: Documentation/installation/index.rst
Comment (joestringer) [inline Documentation/installation/index.rst]: Probably `gs_guide` should point to the actual getting started guides, whereas `gs_install` should point to the installation guides?

**Review by joestringer — APPROVED** [review]
Acknowledged. I understand this was all already reviewed & guided by you, Liz & Bill. 

We *might* still have some links pointing to some of these locations on external sites. We can set up redirects inside ReadTheDocs if necessary. I'm not sure whether we have a good way of detecting 404s when users hit the old pages though. Either way we probably won't find out until these docs changes hit `/stable` after 1.13 is released.

I glanced through anything that isn't just being moved from one place to another and noticed a few nits that might do with some closer attention.

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joestringer) [reply Documentation/gettingstarted/k8s-install-default.rst]: ^Yes

File: Documentation/installation/index.rst
Comment (yoyo-go) [reply Documentation/installation/index.rst:10]: Hi, it seems the [Update Networking, Contributor Guides and Reference sections](https://github.com/cilium/cilium/pull/20642/commits/5f06e740f7a155fc00f837bd0b94d8cf98c723ca) commit is missing.

**Comment by yoyo-go** [comment]
Hi, it seems the [Update Networking, Contributor Guides and Reference sections](https://github.com/cilium/cilium/pull/20642/commits/5f06e740f7a155fc00f837bd0b94d8cf98c723ca) commit is missing.

File: Documentation/index.rst
Comment (qmonnet) [reply Documentation/index.rst:86]: I don't remember the details of that decision, to be honest. I think the idea was to deprecate it because nobody seemed to be using microk8s to run Cilium. @yoyo-go do you remember?

This is probably something we can address as a follow-up anyway.

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (qmonnet) [reply Documentation/gettingstarted/k8s-install-default.rst]: Leftover from a previous naming scheme. I'll fix it, thanks for catching it!

File: Documentation/concepts/observability/index.rst
Comment (qmonnet) [reply Documentation/concepts/observability/index.rst:18]: I don't think we intentionally deleted any of the pages.

`intro` and `hubble-configuration` are now `hubble_intro` and `hubble-configuration` (respectively) under `gettingstarted/`, and they are both listed in the root `Documentation/index.rst`.

Note that Sphinx would complain if some RST documents were not accessible from any of the index.rst, so we should be able to catch it from the CI.

File: Documentation/installation/index.rst
Comment (qmonnet) [reply Documentation/installation/index.rst]: Right, there's even a few bits here and there that needed to be adjusted. I added a new commit (on top of those from the feature branch), would you mind taking a look please?

**Comment by qmonnet** [comment]
> Hi, it seems the [Update Networking, Contributor Guides and Reference sections](https://github.com/cilium/cilium/pull/20642/commits/5f06e740f7a155fc00f837bd0b94d8cf98c723ca) commit is missing.

No, this is not an omission. This commit is not in the feature branch, it has not been reviewed and validated yet (nor is it a straightforward fix for the existing commits). We need to submit it in a separate PR (yours), after this one has been merged.

**Comment by yoyo-go** [comment]
> > Hi, it seems the [Update Networking, Contributor Guides and Reference sections](https://github.com/cilium/cilium/pull/20642/commits/5f06e740f7a155fc00f837bd0b94d8cf98c723ca) commit is missing.
> 
> No, this is not an omission. This commit is not in the feature branch, it has not been reviewed and validated yet (nor is it a straightforward fix for the existing commits). We need to submit it in a separate PR (yours), after this one has been merged.

Got it! Thanks so much!

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (joestringer) [inline Documentation/gettingstarted/k8s-install-default.rst:21]: Does this page describe how to run it specific datapath modes?

Comment (joestringer) [reply Documentation/gettingstarted/k8s-install-default.rst:21]: Ah I guess it's under the "Advanced Installation" section now. :+1:

**Review by joestringer — APPROVED** [review]

File: Documentation/gettingstarted/k8s-install-default.rst
Comment (qmonnet) [reply Documentation/gettingstarted/k8s-install-default.rst:21]: Ah true, I've restored (and renamed) the `gs_install` target for that case. Thanks!

**Comment by qmonnet** [comment]
The changes (except for the last commit) have been reviewed and approved when merging to the feature branch. They have been reviewed again (along with the last commit) by Joe here as well, and documentation tests are passing, so this is ready to merge.

**Pull request merged**
